src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -823,7 +877,7 @@ private void WriteAssembly(int version, ProjectWorld project, BinaryWriter write
                 var obj = new JObject();
                 obj["MessageType"] = "Assembly";
                 obj["ContextId"] = Id;
-                obj[nameof(CompileResponse.Diagnostics)] = ConvertToJArray(project.Diagnostics.Diagnostics);
+                obj[nameof(CompileResponse.Diagnostics)] = ConvertToJArray(project.Diagnostics.CompilationDiagnostics);
```

Comment:
It makes senses because the logic is to construct a `CompileResponse` message out of information gathered during a compilation. __Name-wise__ there is not direct correlation between left and right. What need to be ensure is the content on the right represent the same thing.